client/extensions/woocommerce/app/dashboard/widgets/inventory-widget/index.js

```diff
@@ -0,0 +1,185 @@
+import DashboardWidget from 'woocommerce/components/dashboard-widget';
+import { fetchProducts } from 'woocommerce/state/sites/products/actions';
+import { fetchSettingsProducts } from 'woocommerce/state/sites/settings/products/actions';
+import { getAllProducts } from 'woocommerce/state/sites/products/selectors';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+class InventoryWidget extends Component {
+	static propTypes = {
+		isLoaded: PropTypes.bool.isRequired,
+		isLowStockEnabled: PropTypes.bool.isRequired,
+		lowStockThreshold: PropTypes.number.isRequired,
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+		} ),
+		width: PropTypes.oneOf( [ 'half', 'full' ] ).isRequired,
+	};
+
+	componentDidMount() {
+		const { site } = this.props;
+		if ( site && site.ID ) {
+			this.props.fetchSettingsProducts( site.ID );
+			// Get all products
+			this.props.fetchProducts( site.ID, { offset: 0 } );
+		}
+	}
+
+	getLowStockProducts = () => {
+		const { lowStockThreshold, products } = this.props;
+		const lowProducts = filter( products, p => parseInt( p.stock_quantity ) <= lowStockThreshold );
+		return sortBy( lowProducts, 'stock_quantity' );
+	};
+
+	getClasses = () => {
+		const { lowStockThreshold, noStockThreshold, products } = this.props;
+		const lowProducts = filter( products, p => parseInt( p.stock_quantity ) <= lowStockThreshold );
+		const noProducts = filter( products, p => parseInt( p.stock_quantity ) <= noStockThreshold );
+		return classNames( {
+			'dashboard-widgets__inventory': true,
+			'has-low-stock': lowProducts.length > 0,
+			'has-no-stock': noProducts.length > 0,
+		} );
+	};
+
+	shouldShowImage = () => {
+		const { width } = this.props;
+		const lowProducts = this.getLowStockProducts();
+		// Show the image on full-width, or if there are no products
+		return 'full' === width || ! lowProducts.length;
+	};
+
+	renderRow = ( p, i ) => {
+		const { noStockThreshold, site, translate } = this.props;
+		const outOfStock = parseInt( p.stock_quantity ) <= noStockThreshold;
+		const classes = classNames( 'dashboard-widgets__row', { 'is-out-of-stock': outOfStock } );
+		const link = getLink( `/store/product/:site/${ p.ID }`, site );
+		return (
+			<TableRow key={ i } className={ classes }>
+				<TableItem isTitle>
+					<a href={ link }>{ p.name }</a>
+				</TableItem>
+				<TableItem>{ outOfStock ? translate( 'Out of stock' ) : p.stock_quantity }</TableItem>
+			</TableRow>
+		);
+	};
+
+	renderLowStock = lowProducts => {
+		const { translate } = this.props;
+		const titles = (
+			<TableRow isHeader>
+				<TableItem isHeader isTitle>
+					{ translate( 'Product' ) }
+				</TableItem>
+				<TableItem isHeader isTitle>
+					{ translate( 'Stock' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Stocks' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).